pyiron_base/jobs/job/generic.py

```diff
@@ -1049,6 +1050,9 @@ def to_dict(self):
             "exclude_groups_hdf": self._exclude_groups_hdf,
         }
         data_dict["server"] = self._server.to_dict()
+        self._executable_activate_mpi()
```

Comment:
@pmrv The `self._executable_activate_mpi()` was just moved up the to the point where the `to_dict()` method is called on the `executable` object.